balancer-js/README.md

```diff
@@ -76,7 +76,95 @@ Promise<QueryWithSorOutput {
 }>
 ```
 
-[Example](./examples/queryBatchSwapWithSor.ts)
+### #encodeBatchSwap
+
+Static method to encode a [batch swap](https://dev.balancer.fi/references/contracts/apis/the-vault#batch-swaps).
+
+_NB: This method doesn't execute a batchSwap -- it returns an [ABI byte string](https://docs.soliditylang.org/en/latest/abi-spec.html) containing the data of the function call on a contract, which can then be sent to the network (ex. [sendTransaction](https://web3js.readthedocs.io/en/v1.2.11/web3-eth.html#sendtransaction)). to be executed. See example for more info._
+
+```js
+/**
+ * @param {BatchSwap}           batchSwap - BatchSwap information used for query.
+ * @param {SwapType}            batchSwap.kind - either exactIn or exactOut
+ * @param {BatchSwapSteps[]}    batchSwap.swaps - sequence of swaps
+ * @param {string[]}            batchSwap.assets - array contains the addresses of all assets involved in the swaps
+ * @param {FundManagement}      batchSwap.funds - object containing information about where funds should be taken/sent
+ * @param {number[]}            batchSwap.limits - limits for each token involved in the swap, where either the maximum number of tokens to send (by passing a positive value) or the minimum amount of tokens to receive (by passing a negative value) is specified
+ * @param {string}              batchSwap.deadline -  time (in Unix timestamp) after which it will no longer attempt to make a trade
+ * @returns {string}            encodedBatchSwapData - Returns an ABI byte string containing the data of the function call on a contract
+*/
+Swaps.encodeBatchSwap(batchSwap: {
+    kind: SwapType,
+    swaps: BatchSwapStep[],
+    assets: string[],
+    funds: FundManagement,
+    limits: number[],
+    deadline: string
+}): string
+```
+
+[Example](./examples/batchSwap.ts)
+
+### Swap Service: Flash Swaps
+
+A [Flash Swap](https://dev.balancer.fi/resources/swaps/flash-swaps) is a special type of [batch swap](https://dev.balancer.fi/resources/swaps/batch-swaps) where the caller doesn't need to own or provide any of the input tokens -- the caller is essentiall taking a "flash loan" (an uncollateralized loan) from the Balancer Vault. The full amount of the input token must be returned to the Vault by the end of the batch (plus any swap fees), however any excess of an output tokens can be sent to any address.
+
+IMPORTANT: A "simple" flash swap is an arbitrage executed with only two tokens and two pools,
+swapping in the first pool and then back in the second pool for a profit. For more
+complex flash swaps, you will have to use batch swap directly.
+
+Gotchas:
+
+-   Both pools must have both assets (tokens) for swaps to work
+-   No pool token balances can be zero
+-   flashLoanAmount must not add or subtract > 30% of pool liquidity (see [limitations](https://docs.balancer.fi/v/v1/core-concepts/protocol/limitations#v2-limits))
```

Comment:
The 30% limit is for the original weighted pools. Other pool types will have different limits so it's probably worth removing this.